src/legacy/core_plugins/kbn_vislib_vis_types/public/components/options/metrics_axes/index.tsx

```diff
@@ -0,0 +1,300 @@
+        const lastSeriesAggField = get(lastSeriesAgg, `${matchingSeries[0].id}.field`);
+        const matchingSeriesAggType = get(matchingSeries, '[0]type.name', '');
+        const matchingSeriesAggField = get(matchingSeries, '[0]params.field.name', '');
+
+        const aggTypeIsChanged = lastSeriesAggType !== matchingSeriesAggType;
+        const aggFieldIsChanged = lastSeriesAggField !== matchingSeriesAggField;
+
+        lastMatchingSeriesAgg[matchingSeries[0].id] = {
+          type: matchingSeriesAggType,
+          field: matchingSeriesAggField,
+        };
+        lastLabels[axis.id] = newCustomLabel;
+
+        if (
+          aggTypeIsChanged ||
+          aggFieldIsChanged ||
+          axis.title.text === '' ||
+          lastCustomLabels[axis.id] === axis.title.text
+        ) {
+          // Override axis title with new custom label
+          axes[axisNumber] = { ...axes[axisNumber], title: { ...axis, text: newCustomLabel } };
+          isAxesChanged = true;
+        }
+      }
+    });
+
+    if (isAxesChanged) {
+      setValue('valueAxes', axes);
+    }
+
+    setLastSeriesAgg(lastMatchingSeriesAgg);
+    setLastCustomLabels(lastLabels);
+  };
+
+  const onValueAxisPositionChanged = useCallback(
+    (index: number, value: ValueAxis['position']) => {
+      const valueAxes = [...stateParams.valueAxes];
+      const name = getUpdatedAxisName(value, valueAxes);
+
+      valueAxes[index] = {
+        ...valueAxes[index],
+        name,
+        position: value,
+      };
+      setValue('valueAxes', valueAxes);
+    },
+    [stateParams.valueAxes, getUpdatedAxisName, setValue]
+  );
+
+  const onCategoryAxisPositionChanged = useCallback(
+    (chartPosition: Axis['position']) => {
+      const isChartHorizontal = isAxisHorizontal(chartPosition);
+      setIsCategoryAxisHorizontal(isAxisHorizontal(chartPosition));
+
+      stateParams.valueAxes.forEach((axis, index) => {
+        if (isAxisHorizontal(axis.position) === isChartHorizontal) {
+          const position = mapPosition(axis.position);
+          onValueAxisPositionChanged(index, position);
+        }
+      });
+    },
+    [stateParams.valueAxes, onValueAxisPositionChanged]
+  );
+
+  const addValueAxis = useCallback(() => {
+    const nextAxisIdNumber = stateParams.valueAxes.reduce(
+      countNextAxisNumber(VALUE_AXIS_PREFIX),
+      1
+    );
+
+    const newAxis = cloneDeep(stateParams.valueAxes[0]);
+    newAxis.id = VALUE_AXIS_PREFIX + nextAxisIdNumber;
+    newAxis.position = mapPositionOpposite(newAxis.position);
+    newAxis.name = getUpdatedAxisName(newAxis.position, stateParams.valueAxes);
+
+    setValue('valueAxes', [...stateParams.valueAxes, newAxis]);
+    return newAxis;
+  }, [stateParams.valueAxes, setValue]);
+
+  const removeValueAxis = useCallback(
+    (axis: ValueAxis) => {
+      const newValueAxes = stateParams.valueAxes.filter(valAxis => valAxis.id !== axis.id);
+
+      setValue('valueAxes', newValueAxes);
+
+      let chartIndex;
```

Comment:
It's not super important, but the y axis grid lines setting is also referencing an axis. It should probably also be moved over to an existing axis if it is set

Reply:
You're right. I checked the previous behavior (before `Panel settings` Euification). Y-axis grid lines setting was reseted to `Don't show` value when the chosen axis was removed. I think we should leave that scenario. 
Also now if Y-axis just changes its name (when X-axis position is changed), Y-axis grid lines setting isn't reseted (previously it's also reseted to `Don't show`).

![Sep-10-2019 12-13-27](https://user-images.githubusercontent.com/11524637/64600877-90543800-d3c4-11e9-9278-6d53b0005438.gif)